streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/trait/KubernetesNativeClientTrait.scala

```diff
@@ -49,16 +48,9 @@ trait KubernetesNativeClientTrait extends FlinkClientTrait {
         KubernetesConfigOptions.REST_SERVICE_EXPOSED_TYPE,
         covertToServiceExposedType(submitRequest.k8sSubmitParam.flinkRestExposedType))
 
-    if (submitRequest.buildResult != null) {
-      if (submitRequest.executionMode == ExecutionMode.KUBERNETES_NATIVE_APPLICATION) {
-        val buildResult = submitRequest.buildResult.asInstanceOf[DockerImageBuildResponse]
-        buildResult.podTemplatePaths.foreach(p => {
-          flinkConfig
-            .safeSet(KubernetesConfigOptions.KUBERNETES_POD_TEMPLATE, p._2)
-            .safeSet(KubernetesConfigOptions.JOB_MANAGER_POD_TEMPLATE, p._2)
-            .safeSet(KubernetesConfigOptions.TASK_MANAGER_POD_TEMPLATE, p._2)
-        })
-      }
+    // add flink conf configuration, mainly to set the log4j configuration
+    if (!flinkConfig.contains(DeploymentOptionsInternal.CONF_DIR)) {
+      flinkConfig.safeSet(DeploymentOptionsInternal.CONF_DIR, s"${submitRequest.flinkVersion.flinkHome}/conf")
     }
```

Comment:
From what I understand，KubernetesNativeApplicationClient overwrite the setConfig method, this step won't take effect for k8s application mode submit, maybe we can invoke this method in KubernetesNativeApplicationClient#setConfig ?

Reply:
> From what I understand，KubernetesNativeApplicationClient overwrite the setConfig method, this step won't take effect for k8s application mode submit, maybe we can invoke this method in KubernetesNativeApplicationClient#setConfig ?

Thanks for your review,  updated